src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -147,50 +164,129 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 typeParameters: GetTypeParameters(refactoringResult.TypeToExtractFrom, extractInterfaceOptions.IncludedMembers),
                 members: CreateInterfaceMembers(extractInterfaceOptions.IncludedMembers));
 
-            var interfaceDocumentId = DocumentId.CreateNewId(refactoringResult.DocumentToExtractFrom.Project.Id, debugName: extractInterfaceOptions.FileName);
+            switch (extractInterfaceOptions.Location)
+            {
+                case ExtractInterfaceOptionsResult.ExtractLocation.NewFile:
+                    var containingNamespaceDisplay = GetContainingNamespaceDisplay(refactoringResult.TypeToExtractFrom, refactoringResult.DocumentToExtractFrom.Project.CompilationOptions);
+                    return await ExtractInterfaceToNewFileAsync(
+                        solution: solution,
+                        containingNamespaceDisplay: containingNamespaceDisplay,
+                        extractedInterfaceSymbol: extractedInterfaceSymbol,
+                        refactoringResult: refactoringResult,
+                        extractInterfaceOptions: extractInterfaceOptions,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
dumb question.   why the named parameters here?  They don't seem to add anything (same with the call below)

Reply:
Honestly just a preferencial thing for me, but it made more sense when there were more parameters. I don't mind removing

Reply:
stylistically, it just doesn't match the rest of the IDE.  the majority of our code uses named parameters for one of two cases:

1. the value (usually a literal) doesn't clearly convey what it is.
2. we're eliding optional values, but then need to use the names to properly map the rest of the params.

When a namedParam just matches *exactly* the name of the thing being passed in, that really doesn't help out at all.  i.e. `solution: solution` tells me nothing.  Versus, say `originalSolutionOrWeWillDieAHorribleDeath: solution` :)

Reply:
Yea it doesn't match the other code, so should be fixed. 

Otherwise, I like it because argument order doesn't matter. So when I'm typing out a call, it's just about `solution` needs something, and the local variable I'm using is `solution`. Sometimes the mapping works out like that, sometimes not. I'm not that strongly opinionated about it though, it's just how I got used to typing it out :)